### DISCUSSION TITLE
Save the date that a migration was applied

### DISCUSSION THREAD

**Discussion by lucraraujo** [laravel/framework#43079] [Q&A]
I'm investigating a bug in a application and a possible solution for the bug was DB change made by a migration.
Since the date the migration was created does not matter much I searched for the date the migration was applied. But, since the migration table does not save such data, I wonder if it is there is a place I can go the check this information.
If there is none, I suggest implement a timestamp on the migration table to register the date it was applied.

**Comment by henzeb** [comment]
If you have setup a CI of some sort, you could find the release date?

**Reply by lucraraujo** [reply]
Currently I don't have have a CI. The only thing I have is the tag we create before every deploy in our internal gitlab server.
I know that the reason behind the migrations table is ti keep tabs on which migration was executed, but having the date inside the DB is a simple way to verify when a specific migration was applied.

**Reply by henzeb** [reply]
My suggestion was more to help you find it in the now, not whether this feature would be helpful.

But yes, I think so. Although you can ofcourse alter the table to add updated_at and created_at, filled by MySQL when created or updated. I think that would not break the migration system.

**Comment by erikn69** [comment]
If you are using MySQL or MariaDB you can add a field `created_at` with `CURRENT_TIMESTAMP` as default, so on every new entrie it saves the timestamp

With sql 
```sql
ALTER TABLE migrations ADD COLUMN created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP;
```
Or with migration
```php
$table->timestamp('created_at')->useCurrent();
```

**Comment by lucraraujo** [comment]
I created the following migration:

```php
<?php

declare(strict_types=1);

use Carbon\Carbon;
use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration
{
    public function up(): void
    {
        Schema::table('migrations', static function (Blueprint $table): void {
            $table->dateTime('created_at')->nullable()->default(Carbon::now());
        });
    }

    public function down(): void
    {
        Schema::table('migrations', static function (Blueprint $table): void {
            $table->dropColumn('created_at');
        });
    }
};
```

I work with SQL Server but I can't see a reason that it will not work in other DBs.

**Reply by erikn69** [reply]
I think that not works as expected
Read my answer
